src/components/History/History.tsx

```diff
@@ -1,32 +1,116 @@
-import { ICrossChainAction } from '../../types/crossChainAction';
+import { ICrossChainAction, ICrossChainActionTransaction } from '../../types/crossChainAction';
+
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+//utils
+import { fetchTransactionsData } from '../../utils/transaction';
+import { CHAIN_ID } from '../../utils/chain';
+
+//constants
+import { TRANSACTION_BLOCK_TYPE } from '../../constants/transactionBuilderConstants';
+import { CROSS_CHAIN_ACTION_STATUS } from '../../constants/transactionDispatcherConstants';
+
+import { uniqueId } from 'lodash';
 
 const History = () => {
-  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{ [id: string]: ICrossChainAction[] }>({})
+  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{
+    [id: string]: ICrossChainAction[];
+  }>({});
+  const { getSdkForChainId, accountAddress, sdk } = useEtherspot();
 
   const getLatest = () => {
     try {
-      const storedGroupedCrossChainActionsRaw = getItem(STORED_GROUPED_CROSS_CHAIN_ACTIONS);
-      const storedGroupedCrossChainActionsUpdated = storedGroupedCrossChainActionsRaw
-        ? JSON.parse(storedGroupedCrossChainActionsRaw)
-        : {};
-      setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
+      let storedGroupedCrossChainActionsUpdated;
+      let storedGroupedCrossChainActionsRaw = getItem(STORED_GROUPED_CROSS_CHAIN_ACTIONS);
+      if (storedGroupedCrossChainActionsRaw) {
+        storedGroupedCrossChainActionsUpdated = storedGroupedCrossChainActionsRaw
+          ? JSON.parse(storedGroupedCrossChainActionsRaw)
+          : {};
+        setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
+      } else {
+      }
     } catch (e) {
       //
     }
   };
+  const getTransactionsData = async () => {
+    const sdk = getSdkForChainId(CHAIN_ID.POLYGON);
+    try {
+      if (!sdk) return;
+
+      let results = await fetchTransactionsData(sdk, accountAddress);
+
+      if (results?.items?.length) {
+        results?.items.map((item) => {
+          const chainId = 137;
+          const receiverAddress = item.to;
+          const fromAddress = item.asset.from;
+          const isFromEtherspotWallet = false;
+          const assetUsdPrice = 1;
+          const createTimestamp = +new Date();
+          const crossChainActionId = uniqueId(`${createTimestamp}-`);
+          const transactionId = uniqueId(`${createTimestamp}-`);
+
+          let preview = {
+            chainId,
+            receiverAddress,
+            fromAddress,
+            isFromEtherspotWallet,
+            asset: {
+              address: item.asset.from,
+              decimals: item.asset.decimal,
+              symbol: item.asset.symbol,
+              amount: '1',
+              iconUrl: 'https://polygonscan.com/token/images/klimadao_32.png',
+              usdPrice: assetUsdPrice,
+            },
+          };
```

Comment:
I saw your comment about using static values for testing, but I'm just leaving this comment as a reminder to remove the hardcoded values before merging anything to develop